purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -354,25 +354,34 @@ class Purchases internal constructor(
         }
     }
 
+    /**
+     * Gets the StoreProduct(s) for the given list of product ids for all product types.
+     * @param [productIds] List of productIds
+     * @param [callback] Response callback
+     */
+    fun getProducts(
+        productIds: List<String>,
+        callback: GetStoreProductsCallback
+    ) {
+        getProducts(productIds, null, callback)
+    }
+
     /**
      * Gets the StoreProduct(s) for the given list of product ids.
      * @param [productIds] List of productIds
+     * @param [type] A product type to filter by
      * @param [callback] Response callback
      */
     fun getProducts(
         productIds: List<String>,
+        type: ProductType? = null,
         callback: GetStoreProductsCallback
     ) {
-        getProducts(productIds.toSet(), ProductType.SUBS, object : GetStoreProductsCallback {
-            override fun onReceived(subStoreProducts: List<StoreProduct>) {
-                getProducts(productIds.toSet(), ProductType.INAPP, object : GetStoreProductsCallback {
-                    override fun onReceived(inappStoreProducts: List<StoreProduct>) {
-                        callback.onReceived(subStoreProducts + inappStoreProducts)
-                    }
-                    override fun onError(error: PurchasesError) {
-                        callback.onError(error)
-                    }
-                })
+        val types = type?.let { setOf(type) } ?: setOf(ProductType.SUBS, ProductType.INAPP)
+
+        getProducts(productIds.toSet(), types, object : GetStoreProductsCallback {
```

Comment:
i wonder if we can rename the private versions, just so we have less functions of the same name? i know they're different signatures but it does feel a little harder to read.


maybe something like `getProductsOfType`?